File: SerialReader/SerialReader/PositionalEntry.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SerialReader
{
    class PositionalEntry
    {
        public int EncoderXPos { get; set; }
        public int EncoderYPos { get; set; }
        public IList<DirectionalMeasurement> Measurements = new List<DirectionalMeasurement>();
    }
}

File: UI/src/app/classes/positionalEntry.ts
import {JsonObject, JsonProperty} from "json2typescript";
import { Measurement } from './measurement';
 
@JsonObject("PositionalEntry")
export class PositionalEntry {
    @JsonProperty("EncoderXPos", Number)
    encoderXPos: number = undefined;
    @JsonProperty("EncoderYPos", Number)
    encoderYPos: number = undefined;
    @JsonProperty("Measurements", [Measurement])
    measurements: Measurement[] = undefined;
}

